tests/lib/lazy_orm.py

```diff
@@ -82,9 +87,17 @@ def __call__(self, item):
     def make_orm_class(self, lazy_class):
         attrs = {"__tablename__": lazy_class.__tablename__}
         for name, column_def in lazy_class._columns.items():
-            column = sqlalchemy.Column(*column_def.args, **column_def.kwargs)
+            args, kwargs = self.transform_column_constructor_args(
+                *column_def.args, **column_def.kwargs
+            )
+            column = sqlalchemy.Column(*args, **kwargs)
             attrs[name] = column
         # Construct an ORM class using the supplied base class and the attributes
         # dictionary we've just built
         orm_class = type(lazy_class.__name__, (self.orm_base_class,), attrs)
         return orm_class
+
+    def transform_column_constructor_args(self, type_, *args, **kwargs):
+        # Modify the column type according to entries in the type_map
+        type_ = self.type_map.get(type_, type_)
+        return (type_,) + args, kwargs
```

Comment:
Could you outline a few alternatives that you tried/thought through?

In particular, did you try just reaching in and dynamically modifying the objects?

Reply:
I probably should have discussed with the group earlier: there may well be better solutions floating around. I'll write up my current thoughts in a separate comment on this PR rather than burying them in a reply here.